sql/mozfun/glam/map_from_array_offsets_v2/udf.sql

```diff
@@ -0,0 +1,45 @@
+-- udf_get_values
+CREATE OR REPLACE FUNCTION glam.map_from_array_offsets_v2(
+  required ARRAY<FLOAT64>,
+  `values` ARRAY<FLOAT64>
+)
+RETURNS ARRAY<STRUCT<key STRING, value FLOAT64>> AS (
+  (
+    SELECT
+      ARRAY_AGG(
+        STRUCT<key STRING, value FLOAT64>(
+          CAST(ROUND((key / 10), 1) AS STRING),
+          `values`[OFFSET(SAFE_CAST(key AS INT64))]
+        )
+        ORDER BY
+          key
+      )
+    FROM
+      UNNEST(required) AS key
+  )
+);
+
+SELECT
+  assert.array_equals(
+    ARRAY<STRUCT<key STRING, value FLOAT64>>[("0", 2.0), ("2", 4.0)],
+    glam.map_from_array_offsets([0.0, 2.0], [2.0, 3.0, 4.0])
```

Comment:
These tests should be calling `glam.map_from_array_offsets_v1` or whatever the new function name is. Is this the issue you had mentioned about permissions? I thought you meant it was being called from somewhere else. In the tests it should work. If it's not working, you may need to authenticate.